src/main/java/model/FruitTransaction.java

```diff
@@ -0,0 +1,53 @@
+package model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public static Operation fromCode(String code) {
+            for (Operation op : values()) {
+                if (op.code.equals(code)) {
+                    return op;
+                }
+            }
+            throw new IllegalArgumentException("Unknown operation: " + code);

```

Comment:
The method 'fromCode' in the Operation enum could potentially return a null value if none of the codes match. It's better to handle this case explicitly by throwing an exception, which you have done. However, consider adding a null check for the input 'code' to avoid a NullPointerException.